### PR TITLE
Ports Floor Painter From Paradise

### PR DESCRIPTION
**Pull request by AndrewMontagne**
gif

Ported the floor painter in all its glory, improved the tile rotation UX and made it need toner to use for good measure. You can paint most, if not all of the bar without running out.

Also added colour variations for airlock painter, floor painter and pipe painter sprites. Not much, but enough to tell them apart in your belt:

image
*Airlock painter, floor painter and pipe painter respectively.*

Also fixed some errors I made with the floor sprites that I made in #271: 
- Incorrect colours for grout.
- The wrong shade of black for the "dark" set of tiles
- Leftover duplicate-named tiles which conflicted in the dmi
- Changed the "caution" tiles to use the hazard tape sprites

I also reduced the pipe painter manufacture cost in line with the airlock painter, because the cost difference was just dumb.

:cl: AndrewMontagne + Paradise
add: Floor Painter
imageadd: Colour variations for airlock painter, floor painter and pipe painter sprites
tweak: Tweaked some defects on floor sprites
tweak: Reduced cost for pipe painter
fix: Scorched tile sprites having broken scorch states
/:cl:

### CODE REVIEW

**Comment by Cactus2388** [comment]
Is that andrewUI? looks like it.

**Comment by AndrewMontagne** [comment]
@Cactus2388 No, and stop calling it that. >:|

**Comment by AffectedArc07** [comment]
I was gonna ask that question. I would make it have more toner capacity though.

File: code/game/objects/items/devices/floor_painter.dm [hunk 1]
```diff
@@ -0,0 +1,201 @@
+// Floor painter
+
+/obj/item/device/floor_painter
+	name = "floor painter"
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "paint sprayer2"
```
Comment (FlattestGuitar) [inline code/game/objects/items/devices/floor_painter.dm]: Not a fan of all these space-including icon states. While I think they should be fine, the convention is generally using underscores. Please.

File: code/game/objects/items/devices/floor_painter.dm [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	else
+		return ..()
+
+/obj/item/device/floor_painter/examine(mob/user)
+	..()
+	if(!ink)
+		to_chat(user, "<span class='notice'>It doesn't have a toner cartridge installed.</span>")
+		return
+	var/ink_level = "high"
+	if(ink.charges <= 0)
```
Comment (FlattestGuitar) [inline code/game/objects/items/devices/floor_painter.dm]: should check against `charges_per_use`, not 0

File: code/game/objects/items/devices/floor_painter.dm [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	to_chat(user, "<span class='notice'>Its ink levels look [ink_level].</span>")
+
+/obj/item/device/floor_painter/afterattack(var/atom/A, var/mob/user, proximity, params)
+	if(!proximity)
+		return
+
+	if(!ink)
+		to_chat(user, "<span class='notice'>There is no toner cartridge installed in [src]!</span>")
+		return FALSE
+	else if(ink.charges <= 0)
```
Comment (FlattestGuitar) [inline code/game/objects/items/devices/floor_painter.dm]: same here, `charges_per_use`

**Pull request merged**